snapcraft/internal/meta/plugs.py

```diff
@@ -54,15 +54,16 @@ def validate(self) -> None:
     def from_dict(cls, *, plug_dict: Dict[str, Any], plug_name: str) -> "Plug":
         """Create plug from dictionary."""
 
-        interface = plug_dict.get("interface", None)
-        if interface is not None:
-            plug_class = PLUG_MAPPINGS.get(interface, None)
-            if plug_class is not None:
-                return plug_class.from_dict(plug_dict=plug_dict, plug_name=plug_name)
+        plug_dict = {} if plug_dict is None else plug_dict
+        interface = plug_dict.get("interface", plug_name)
```

Comment:
Is this redundant given the change to `Plug.__init__`?

Reply:
Indeed

Reply:
No - "`from_dict`" is a second constructor by the "`@classmethod`" decorator.

If we use, the classic "`if`" assignment, then it would become clearer. 



```
        if plug_dict is not None:
            interface = plug_dict.get("interface", plug_name)
            plug_class = PLUG_MAPPINGS.get(interface, None)
            if plug_class is not None:
                return plug_class.from_dict(plug_dict=plug_dict, plug_name=plug_name)
```

But here the indention-level increases a bit.